core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -104,13 +104,25 @@ sealed abstract class Aggregate extends Data {
 
   private[chisel3] def width: Width = getElements.map(_.width).foldLeft(0.W)(_ + _)
 
-  private[chisel3] def legacyConnect(that: Data)(implicit sourceInfo: SourceInfo): Unit = {
+  // Emits the FIRRTL `this <= that`, or `this is invalid` if that == DontCare
+  private[chisel3] def firrtlConnect(that: Data)(implicit sourceInfo: SourceInfo): Unit = {
+    // If the source is a DontCare, generate a DefInvalid for the sink, otherwise, issue a Connect.
+    if (that == DontCare) {
+      pushCommand(DefInvalid(sourceInfo, lref))
+    } else {
+      pushCommand(Connect(sourceInfo, lref, Node(that)))
+    }
+  }
+
+  // Emits the FIRRTL `this <- that`, or `this is invalid` if that == DontCare
+  // Note that `this <= that` will be emitted if both `this` and `that` are not aggregate types
```

Comment:
Is there a good reason to special case this situation as opposed to letting a FIRRTL compiler do whatever it wants with it? I think that connect and partial connect both have the same behavior here?

Reply:
Hmm. I can't remember, does partial connect error if lhs is a source flow, or silently do nothing?

Reply:
But the original reason why I special cased it is to match what Chisel was doing before this PR.

Reply:
I think we aren't special casing anymore, so clarified the comment.